src/iris/api.py

```diff
@@ -5604,6 +5610,303 @@ def on_get(self, req, resp):
         resp.body = ujson.dumps(payload)
 
 
+class SenderHeartbeat():
+    allow_read_no_auth = False
+    internal_allowlist_only = True
+
+    def __init__(self, config):
+        cfg = config.get("iris-message-processor", {})
+        self.zk_hosts = cfg.get("zk_hosts", "127.0.0.1:2181")
+        self.lock_path = cfg.get("lock_path", "/iris_message_processor/db_locks/bucket_lock")
+        self.number_of_buckets = cfg.get("number_of_buckets", 100)
+        self.sender_ttl = config.get("sender_ttl", 60)
+        self.zk_debug = config.get("zk_debug", True)
+
+    def on_get(self, req, resp, node_id):
+
+        start = datetime.datetime.now()
+        payload = {}
+
+        # configure sql client
+        connection = db.engine.raw_connection()
+        cursor = connection.cursor(db.dict_cursor)
+
+        if self.zk_debug:
+            # use exitstack to replace zookeeper lock context manager for testing
+            lock = ExitStack()
+        else:
+            # configure zookeeper client
+            zk = KazooClient(self.zk_hosts)
```

Comment:
It should be possible to do so. I think it shouldn't cause issues with the lock either since it is not re-entrant and repeat calls will block. That being said I was planning on making that change alongside my cache change since they are similar refactoring and this PR is already quite large and complicated.